### PR TITLE
Appply some refurb suggestions

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**

### CODE REVIEW

**Comment by brettcannon** [comment]
Thanks for the PR, but it feels a bit noisy, e.g. changing lists to tuples that don't need to be changed. I'm going to close this, but if you think there's something I'm missing then please let us know or open a more targeted PR.

File: src/packaging/requirements.py [hunk 1]
```diff
@@ -38,7 +38,7 @@ def __init__(self, requirement_string: str) -> None:
 
         self.name: str = parsed.name
         self.url: Optional[str] = parsed.url or None
-        self.extras: Set[str] = set(parsed.extras if parsed.extras else [])
+        self.extras: Set[str] = set(parsed.extras or [])
```
Comment (DimitriPapadopoulos) [inline src/packaging/requirements.py:41]: You should at least keep this one.

File: tests/test_specifiers.py [hunk 2]
```diff
@@ -235,8 +235,7 @@ def test_specifiers_hash(self, specifier):
 
     @pytest.mark.parametrize(
         ("left", "right", "op"),
-        itertools.chain(
-            *
+        itertools.chain.from_iterable(
```
Comment (DimitriPapadopoulos) [inline tests/test_specifiers.py:238]: The following series of `chain.from_iterable` looks interesting to me as well.

**Review by DimitriPapadopoulos — COMMENTED** [review]
@brettcannon I can understand for changing lists to tuples, perhaps this one should be left out. The rest of the suggestions look like good ideas. The idea was to silence more ruff issue before adding a couple rules to jaraco/skeleton, for consistency across pypa projects. flake8-bugbear (B) rules are arguably more interesting, the tool has found two errors in the test code.

**Comment by brettcannon** [comment]
> flake8-bugbear (B) rules are arguably more interesting, the tool has found two errors in the test code.

Then please open another PR w/ just those changes.

**Pull request closed without merging**